staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -397,11 +399,11 @@ func (o *ApplyOptions) Run() error {
 			}
 
 			helper := resource.NewHelper(info.Client, info.Mapping)
-			if o.ServerDryRun {
+			if o.DryRunStrategy == cmdutil.DryRunServer {
 				if err := resource.VerifyDryRun(info.Mapping.GroupVersionKind, o.DynamicClient, o.DiscoveryClient); err != nil {
 					return err
 				}
-				helper.DryRun(o.ServerDryRun)
+				helper.DryRun(true)
```

Comment:
Yeah, that's the part that was missing before, but now that the variable is gone, there is no way to miss this.